zip-0226.rst

```diff
@@ -325,6 +325,111 @@ A ZSA Asset Burn description is encoded in a transaction as an of a AssetBurn ty
 | 8     | valueBurn | :math:`\{1 .. 2^{64} - 1\}` | The amount of the AssetType being burnt.                                                                                  |
 +-------+-----------+-----------------------------+---------------------------------------------------------------------------------------------------------------------------+
 
+TxId Digest
+===========
+
+The transaction digest algorithm defined in ZIP 244 [#zip-0244]_ is modified by the ZSA protocol to add a new branch for issuance information, and add an additional hash for the Asset Base within the ``orchard_digest``. The details of these changes are described in this section. The overall tree structure of the hash is as follows, and the additional names not referenced in ZIP 244 are described in detail below::
+
+    txid_digest
+    ├── header_digest
+    ├── transparent_digest
+    │   ├── prevouts_digest
+    │   ├── sequence_digest
+    │   └── outputs_digest
+    ├── sapling_digest
+    │   ├── sapling_spends_digest
+    │   │   ├── sapling_spends_compact_digest
+    │   │   └── sapling_spends_noncompact_digest
+    │   ├── sapling_outputs_digest
+    │   │   ├── sapling_outputs_compact_digest
+    │   │   ├── sapling_outputs_memos_digest
+    │   │   └── sapling_outputs_noncompact_digest
+    │   └── valueBalance
+    └── orchard_digest
+    │   ├── orchard_actions_compact_digest
+    │   ├── orchard_actions_assetid_digest
+    │   ├── orchard_actions_memos_digest
+    │   ├── orchard_actions_noncompact_digest
+    │   ├── flagsOrchard
+    │   ├── valueBalanceOrchard
+    │   └── anchorOrchard
+    └── issuance_digest
+        ├── issue_actions_digest
+        └── issuanceValidatingKey
+
+In the specification below, nodes of the tree are presented in depth-first order.
+
+txid_digest
+-----------
+A BLAKE2b-256 hash of the following values ::
+
+   T.1: header_digest       (32-byte hash output)
+   T.2: transparent_digest  (32-byte hash output)
+   T.3: sapling_digest      (32-byte hash output)
+   T.4: orchard_digest      (32-byte hash output)
+   T.5: issuance_digest     (32-byte hash output)
+
+The personalization field remains the same as in ZIP 244 [#zip-0244]_. The branches ``T.1: header_digest``, ``T.2: transparent_digest``, and ``T.3: sapling_digest`` are as defined in ZIP 244.
+
+T.4: orchard_digest
+```````````````````
+When Orchard Actions are present in the transaction, this digest is a BLAKE2b-256 hash of the following values ::
+
+   T.4a: orchard_actions_compact_digest      (32-byte hash output)
+   T.4b: orchard_actions_assetid_digest      (32-byte hash output)
```

Comment:
Not sure where this came from. There is no explicit `assetid` in an `action`. All assetid information is inside the encrypted note and should not be hashed separately. right?

Reply:
Removed.